### PR TITLE
ARC-2131 - unified logging for backfill tasks

### PR DESCRIPTION
**Pull request by joshkay10**
**What's in this PR?**
Making all the start/complete logging for the 5 backfill tasks the same!!!

**Why**
Makes Splunking a lot easier, and it give us way easier debugging when trying to follow the backfill.

You can now search Splunk with "Backfill task *" and you will see when each task started and completed and whether there was data. JiraHost, messageId and githubInstallationId to filter it further.

**Whats Next?**
Probably a small runbook to help

### CODE REVIEW

**Review by sasank-xyz — APPROVED** [review]

File: src/sync/branches.ts
Comment (gxueatlassian) [inline src/sync/branches.ts:20]: Put it as `Branch backfill task started`? As if just `Backfill task started` it is a bit confusing.
And also I don't think we need this line, as we should have already something in a upper level? In the coordinator of all the tasks? So instead of doing it in each, leave it to parent?

File: src/sync/branches.ts
Comment (gxueatlassian) [inline src/sync/branches.ts:29]: Same as above, need `Branch xxx` in the msg. Also not sure we need this line.

File: src/sync/branches.ts
Comment (gxueatlassian) [inline src/sync/branches.ts:17]: The app already has in the parent sync/installation.ts. I feel this line is like duplication. 
```
		const logger = parentLogger.child({
				task: nextTask
			});

			logger.info("Starting task");
			nextTask.startTime = Date.now();
```

File: src/sync/branches.ts
Comment (gxueatlassian) [inline src/sync/branches.ts:29]: Again, that `{ processingTime: Date.now() - startTime` should be somewhere upper level in sync/installation.ts
However I understand why here, as we need some more logging for `jiraPayload?.branches?.length`. 

I want to understand the purpose of the changes here. If it is for auditing problem, we might need to put more info in it. If it is for performance reason, we probably should use statsD instead.

File: src/sync/branches.ts
Comment (joshkay10) [reply src/sync/branches.ts:17]: It does, but theres a few steps between that log and the actual execution, the parent log here is to tag all the other logs in the function a bit better, the `task:nextTask` attribute we inherit is an object and its a pain to Splunk with.

Otherwise its pretty much a tidier log :shrug:

File: src/sync/branches.ts
Comment (joshkay10) [reply src/sync/branches.ts:29]: You are correct! the step after this is to emit that processingTime with StatsD, whether it needs to be in the log or not.... maybe/maybe not but it was floating around :lol: , this log was mainly just to see where it returns, so we can trace the flow if we need to.

Same as before, its an existing log, just aligning them across all the tasks

File: src/sync/branches.ts
Comment (gxueatlassian) [reply src/sync/branches.ts:17]: Inside the `const taskExecutor = async (nextTask: Task` in the parent ( as in my above comments ), the `logger.info("Starting task");` actually means the start of the task, I don't see any extra steps?

Comment (gxueatlassian) [reply src/sync/branches.ts:17]: I'll mark this as /nit for now to unblock you. But let me know if you can think of anyway to improve.

**Review by gxueatlassian — APPROVED** [review]

**Pull request merged**
